plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -459,7 +464,7 @@ public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, Li
                         throw new UnsupportedOperationException("Unsupported sort order: " + sortItem.getSortOrder());
                     })
                     .collect(joining(", "));
-            return format("%s ORDER BY %s OFFSET 0 ROWS FETCH NEXT %s ROWS ONLY", query, orderBy, limit);
+            return format("SELECT TOP (%d) %s ORDER BY %s", limit, query.substring(start.length()), orderBy);
```

Comment:
I think the previous impl is as effective, and less tricky (no SQL pattern matching needed)

Reply:
Fair point.